tools/istio-iptables/pkg/cmd/root.go

```diff
@@ -136,6 +136,12 @@ func bindCmdlineFlags(cfg *config.Config, cmd *cobra.Command) {
 		&cfg.NetworkNamespace)
 
 	flag.BindEnv(fs, constants.CNIMode, "", "Whether to run as CNI plugin.", &cfg.CNIMode)
+
+	flag.BindEnv(fs, constants.NoReconcile, "", "Skip reconciliation of pre-existing and incompatible iptables rules, failing instead if drift detected",
```

Comment:
I am not sure I fully understand. Looks like we did all of this work, and then turned it off. Are there plans to turn it on by default once its more tested? Is it supposed to be on-by-default for VMs?

Reply:
The reconciliation is currently disabled for sidecars and enabled for all other use cases. This decision was made because (1) reconciliation is not really relevant for sidecars, and (2) to adopt a safer approach in rolling out this feature.
In the case of sidecars, we don't need to handle the transition from old rules to new rules because if we were to change the args of istio-iptables, Kubernetes would simply recreate the pod. We just need to have checks to prevent reapply rules if existing rules are found and they are the same as the ones the rerun would apply.
Therefore, the behavior for sidecars that this commit introduced is as follows:

- If no "leftover" rules/chains are found (first run): Create rules/chains as usual. Exit 0.
- If "leftovers" compatible with the planned rules/chains are found: Skip applying new rules/chains and print some logs. Exit 0.
- If incompatible "leftovers" are found: Print logs and exit 1. If reconciliation were enabled, then this scenario would have led to: apply_guardrails -> remove_old_rules -> apply_new_rules -> remove_guardrails.

Scenario 3 would only occur in sidecars if the initial istio-init execution was unsuccessful, leading to incomplete rules/chains (a very niche scenario).